azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -86,12 +98,115 @@ public ProjectManager(ProjectLoader loader, Props props) {
     new XmlValidatorManager(prop);
     loadAllProjects();
     loadProjectWhiteList();
+
+    CleanerThread cleanerThread = new CleanerThread();
+    cleanerThread.start();
+  }
+
+  /**
+   * Cleaner thread to clean projects. Runs every day.
+  **/
+  private class CleanerThread extends Thread {
+
+    // runs every day
+    private static final long CLEANER_THREAD_WAIT_INTERVAL_MS = 24 * 60 * 60 * 1000;
```

Comment:
Shouldn't this be configurable as well ?

Reply:
We can do this as configurable as well, in other places, it was hardcoded so did this way, you can tell me if we want to make this configurable, I will do the necessary changes. This was not looking as a frequent change